recognition/include/pcl/recognition/impl/linemod/line_rgbd.hpp

```diff
@@ -43,176 +43,143 @@
 #include <fcntl.h>
 #include <pcl/point_cloud.h>
 #include <limits>
+#include <boost/filesystem.hpp>
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template <typename PointXYZT, typename PointRGBT> bool
-pcl::LineRGBD<PointXYZT, PointRGBT>::readLTMHeader (int fd, pcl::io::TARHeader &header)
+pcl::LineRGBD<PointXYZT, PointRGBT>::loadTemplates(const std::string& file_name,
+                                                   const std::size_t object_id)
 {
-  // Read in the header
-  int result = static_cast<int> (::read (fd, reinterpret_cast<char*> (&header.file_name[0]), 512));
-  if (result == -1)
-    return (false);
-
-  // We only support regular files for now. 
-  // Additional file types in TAR include: hard links, symbolic links, device/special files, block devices, 
-  // directories, and named pipes.
-  if (header.file_type[0] != '0' && header.file_type[0] != '\0')
-    return (false);
+  namespace fs = boost::filesystem;
 
-  // We only support USTAR version 0 files for now
-  if (std::string (header.ustar).substr (0, 5) != "ustar")
+  fs::path path_to_file (file_name);
+  if (!fs::exists (path_to_file) || !(path_to_file.extension () == ".ltm"))
+  {
+    PCL_ERROR ("[pcl::LineRGBD::loadTemplates] Input file is not valid. Must exist and have .ltm file extension!\n");
     return (false);
+  }
 
-  if (header.getFileSize () == 0)
+  const auto parent_dir = path_to_file.parent_path ();
+  const auto temp_dir = parent_dir.string () + fs::path::preferred_separator + "tmp";
```

Comment:
Why not use `boost::filesystem::temp_directory_path()`?

Reply:
that's a possibility as well. I just thought creating a `tmp` as sub-dir to the provided path is more convenient for clean up.

Reply:
We might not have permissions to write in the location

Reply:
ok, i'll edit this part